src/parcsr_ls/par_amg_setup.c

```diff
@@ -302,6 +303,22 @@ hypre_BoomerAMGSetup( void               *amg_vdata,
 
    grid_relax_type[3] = hypre_ParAMGDataUserCoarseRelaxType(amg_data);
 
+   /* Get the number of vector components when LHS/RHS are passed in */
+   if ((f != NULL) && (u != NULL))
+   {
+      /* Verify that the number of vectors held by f and u match */
+      if (hypre_ParVectorNumVectors(f) != hypre_ParVectorNumVectors(u))
+      {
+         hypre_error_w_msg(HYPRE_ERROR_GENERIC, "Error: num_vectors for RHS and LHS do not match!\n");
+         return hypre_error_flag;
+      }
+      num_vectors = hypre_ParVectorNumVectors(f);
+   }
+   else
+   {
+      num_vectors = 1;
```

Comment:
Can we pass in `NULL` `f` or `u`? Just curious.

Reply:
AMS does that in a couple of places, e.g.:
https://github.com/hypre-space/hypre/blob/6425099995732e50b816ebd228e831e21b000f80/src/parcsr_ls/ams.c#L3211-L3213

https://github.com/hypre-space/hypre/blob/6425099995732e50b816ebd228e831e21b000f80/src/parcsr_ls/ams.c#L3294-L3296